docs/contributing/inheritance.md

```diff
@@ -0,0 +1,87 @@
+# Stravalib's Inheritance Patterns
+
+Stravalib's API is built directly from Strava's `swagger.json` example response
+data. We use the Pydantic `BaseModel` combined with `datacodegen` to build our
+base model objects (found in `strava_model.py`), which we then enhance and
+update to align with the data actually returned by the Strava API.
+
+This page will help you understand Stravalib's inheritance patterns.
+
+In our experience maintaining Stravalib, the Strava API online specification
+and the `swagger.json` file don't always align perfectly with the data actually
+returned by Strava. As a result, you will notice that we frequently overwrite
+data types to match the actual data we see returned from the Strava API for
+each endpoint. [More information on this process can be found here.](ci_api_updates)
+
+Below, we present the inheritance schema for Stravalib.
+
+## Inheritance overview
+
+At a high level, there are two modules:
+
+<!-- stravalib.model link doesn't work but strava_model does not.
+I'm not sure why. We can fix this later once we get inheritance diagrams working as we expect them to. There is something funky with how I set up api autodoc.
+-->
+* {py:mod}`stravalib.model` and
+* {py:mod}`stravalib.strava_model`.
+
+`strava_model` is generated automatically using a CI build from the Strava API.
+Stravalib uses the model.py module to do a few things:
+
+1. It supports inheritance of the {py:class}`stravalib.model.BoundClientEntity`, which supports API calls for lazily loaded properties,
+2. it allows us to override {py:mod}`stravalib.strava_model` attributes that have typed attributes that don't align with the actual API responses, and
+3. allows us to add attributes that are found in the returned data but not documented or found in the swagger.json response.
+
+The full inheritance pattern, which includes inheritance from both `strava_model` and `pydantic.BaseModel`, is below.
+
+### Inheritance diagram showing the relationship between strava_model.py and model.py
+
+:::{mermaid}
+
+classDiagram
+    %% Activities in model namespace
```

Comment:
Can we try the following, so the most abstract class is shown on top (which is more or less standard)?
```suggestion
classDiagram
    direction BT
    %% Activities in model namespace
```

Reply:
absolutely.